### PR TITLE
Python integration

### PR DESCRIPTION
**Pull request by HLWeil**
- Setup `build` and `test` targets for `Python`
- Full integration of Fable transpilation to `Python`
- Extended developer setup instructions
- Moved JsonSchemaValidation to tests

### CODE REVIEW

**Comment by HLWeil** [comment]
CI needs to be updated

File: build/GenerateIndexPy.fs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        ".Templates.template", "Template"
+        ".Templates.templates", "Templates"
+        ".Templates.template", "Organisation"
+        ".arc","ARC"
+    |]
+
+let ARCtrl_generate (rootPath: string) = 
+    generateIndexFileContent classes
+    |> Array.reduce (fun a b -> a + "\n" + b)
+    |> writePyIndexfile rootPath
```
Comment (Freymaurer) [inline build/GenerateIndexPy.fs:49]: 🚀

File: build/ProjectInfo.fs [hunk 2]
```diff
@@ -30,8 +30,9 @@ let gitHome = $"https://github.com/{gitOwner}"
 
 let projectRepo = $"https://github.com/{gitOwner}/{project}"
 
-let pkgDir = "dist/pkg"
+let netPkgDir = "dist/pkg"
```
Comment (Freymaurer) [inline build/ProjectInfo.fs:33]: no relative path?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+[tool.poetry]
+name = "ARCtrl"
+version = "1.0.0"
+description = "Library for management of Annotated Research Contexts (ARCs) using an in-memory representation and runtimer agnostic contract systems."
+authors = ["Heinrich Lukas Weil <weil@rptu.de>"]
```
Comment (Freymaurer) [inline pyproject.toml:5]: freymaurer@gmx.de

File: pytests/arc_table_tests.ipynb
Comment (Freymaurer) [inline pytests/arc_table_tests.ipynb:1]: Should this file be on .gitignore?

File: pytests/composite_header_tests.ipynb
Comment (Freymaurer) [inline pytests/composite_header_tests.ipynb:1]: Same

File: README.md [hunk 4]
```diff
@@ -44,10 +44,27 @@ Currently we provide some documentation in form of markdown files in the `/docs`
     - verify with `npm --version` (Tested with v9.2.0)
 - [.NET SDK](https://dotnet.microsoft.com/en-us/download)
     - verify with `dotnet --version` (Tested with 7.0.306)
+- [Python](https://www.python.org/downloads/)
+    - verify with `py --version` (Tested with 3.12.2)
```
Comment (Freymaurer) [inline README.md:48]: Add a note for minimal 3.11.x

File: src/ARCtrl/WebRequest/WebRequest.Py.fs [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+#if FABLE_COMPILER_PYTHON
+type private InteropResponseType =
+    abstract status_code : int
+    abstract text : string
+    abstract encoding : string
+    abstract headers : obj
+
+[<RequireQualifiedAccess>]
+module Interop =
```
Comment (Freymaurer) [inline src/ARCtrl/WebRequest/WebRequest.Py.fs:17]: maybe we should do the same for the web request library from Zaid (copy paste the relevant code). Then we could implement the nodejs part in a cleaner way.

**Review by Freymaurer — CHANGES_REQUESTED** [review]

File: pytests/arc_table_tests.ipynb
Comment (HLWeil) [reply pytests/arc_table_tests.ipynb:1]: Removed the whole folder and setup file. Was there for initial testing

File: src/Contract/Contract.fs [hunk 6]
```diff
@@ -70,7 +74,9 @@ type Contract =
         /// The actual DTO, as discriminate union.
         DTO: DTO option
     }
+    #if FABLE_COMPILER_JAVASCRIPT
     [<NamedParams(fromIndex=2)>]
```
[the remark this answers is not in the document]
Comment (HLWeil) [reply src/Contract/Contract.fs:78]: Yup, there were some problems. Will make a note

File: testpy.cmd
[the remark this answers is not in the document]
Comment (HLWeil) [reply testpy.cmd:1]: Removed

File: tests/ISA/ISA.Json.Tests/ArcTypes.Tests.fs [hunk 7]
```diff
@@ -282,9 +282,19 @@ let tests_ArcAssay = testList "ArcAssay" [
             for j = 0 to cells.Length - 1 do
                 t.Values.[(j,i)] <- cells.[j]
         let f() = ArcAssay.toJsonString a
+        #if FABLE_COMPILER_JAVASCRIPT
```
[the remark this answers is not in the document]
Comment (HLWeil) [reply tests/ISA/ISA.Json.Tests/ArcTypes.Tests.fs:285]: Will move all of this into a performance checking project anyways.

File: src/Contract/Contract.fs (hunk 6 above)
[the remark this answers is not in the document]
Comment (Freymaurer) [reply src/Contract/Contract.fs:78]: yeah exactly 🙂

File: src/ISA/ISA/Fable.fs [hunk 8]
```diff
@@ -118,4 +118,35 @@ let distinct_generic l1 =
...
-    |> Seq.reduce (fun a b -> a + b)
\ No newline at end of file
+    |> Seq.reduce (fun a b -> a + b)
+
+module Array = 
+
+    
+    let sort (a : 'T []) = 
+         #if FABLE_COMPILER_PYTHON
+             a |> List.ofArray |> List.sort |> Array.ofList
```
[the remark this answers is not in the document]
Comment (HLWeil) [reply src/ISA/ISA/Fable.fs:128]: https://github.com/fable-compiler/Fable/issues/3638

Seems to be fixed now.

File: build/release_package.json [hunk 9]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@nfdi4plants/arctrl",
-  "version": "1.1.0+6309e03",
+  "version": "1.2.0+19d850e",
```
Comment (Freymaurer) [inline build/release_package.json:3]: use short semver without metadata

**Review by Freymaurer — CHANGES_REQUESTED** [review]

**Comment by HLWeil** [comment]
Done, will be updated with next version tick.

**Review by Freymaurer — APPROVED** [review]
lets go! 🚀

**Pull request merged**